packages/hardhat-core/src/internal/hardhat-network/provider/node.ts

```diff
@@ -926,6 +928,76 @@ Hardhat Network's forking functionality only works with blocks from at least spu
     await this._txPool.updatePendingAndQueued();
   }
 
+  public async traceTransaction(
+    hash: Buffer,
+    config: RpcDebugTracingConfig
+  ): Promise<object> {
+    const block = await this.getBlockByTransactionHash(hash);
+    if (block === undefined) {
+      throw new InvalidInputError(
+        `Unable to find a block containing transaction ${bufferToHex(hash)}`
+      );
+    }
+
+    return this._runInBlockContext(
+      new BN(block.header.number).subn(1),
+      async () => {
+        const blockNumber = block.header.number.toNumber();
+        const blockchain = this._blockchain;
+        let vm = this._vm;
+        if (
+          blockchain instanceof ForkBlockchain &&
+          blockNumber <= blockchain.getForkBlockNumber().toNumber()
+        ) {
+          const common = getCommon(blockchain.getNetworkId(), blockNumber);
+
+          vm = new VM({
+            common,
+            activatePrecompiles: true,
+            stateManager: this._vm.stateManager,
+            blockchain: this._vm.blockchain,
+          });
+        }
+
+        const isPreSpuriousDragon = !vm._common.gteHardfork("spuriousDragon");
```

Comment:
I've already forgotten the reason for this. We should add a comment explaning it.

Reply:
Added a comment on 6fae2f732112c14b4830d3558f9027d8e967d8b3. I _think_ that was the reason.